src/cript/api/api.py

```diff
@@ -179,218 +305,200 @@ def is_vocab_valid(
-        A context manager is preferred where possible.
-        Jupyter notebooks are a use case where this connection can be handled manually.
-        If this function is called manually, the `API.disconnect` function has to be called later.
-
-        For manual connection: nested API object are discouraged.
-        """
-        # Store the last active global API (might be None)
-        global _global_cached_api
-        self._previous_global_cached_api = copy.copy(_global_cached_api)
-        _global_cached_api = self
-        return self
-
-    def disconnect(self):
-        """
-        Disconnect this API from the active access point.
-        It is not necessary to call this function manually if a context manager is used.
-        A context manager is preferred where possible.
-        Jupyter notebooks are a use case where this connection can be handled manually.
-        This function has to be called manually if  the `API.connect` function has to be called before.
-
-        For manual connection: nested API object are discouraged.
-        """
-        # Restore the previously active global API (might be None)
-        global _global_cached_api
-        _global_cached_api = self._previous_global_cached_api
+        db_schema = self._get_db_schema()
 
-    def __enter__(self):
-        self.connect()
+        node_dict = json.loads(node_json)
+        try:
+            node_list = node_dict["node"]
+        except KeyError:
+            raise CRIPTNodeSchemaError(error_message=f"'node' attriubte not present in serialization of {node_json}. Missing for exmaple 'node': ['material'].")
 
-    def __exit__(self, type, value, traceback):
-        self.disconnect()
+        if isinstance(node_list, list) and len(node_list) == 1 and isinstance(node_list[0], str):
+            node_type = node_list[0]
+        else:
+            raise CRIPTJsonNodeError(node_list, node_type)
 
-    @property
-    def host(self):
-        """
-        Read only access to the currently connected host.
-        If the connection to a new host is desired, create a new API object.
-        """
-        return self._host
+        # set which node you are using schema validation for
+        db_schema["$ref"] = f"#/$defs/{node_type}Post"
 
-    def save(self, node: PrimaryBaseNode) -> None:
-        """
-        This method takes a primary node, serializes the class into JSON
-        and then sends the JSON to be saved to the API
+        try:
+            jsonschema.validate(instance=node_dict, schema=db_schema)
+        except jsonschema.exceptions.ValidationError as error:
+            raise CRIPTNodeSchemaError(error_message=str(error))
 
-        Parameters
-        ----------
-        node: primary node
-            the Primary Node that the user wants to save
+        # if validation goes through without any problems return True
+        return True
 
-        Returns
-        -------
-        None
+    def save(self, project: Project) -> None:
```

Comment:
See #77 of how I disagree here.

Reply:
To me personally `api.save(my_project)` just makes more sense to me than `my_project.save()`, but if we want to move to that then we can do that as well. I'm thinking you and other scientists are going to be using the SDK the most, so whatever you guys think is best we can do that.

This might be a good place to use our [poll feature](https://github.com/C-Accel-CRIPT/Python-SDK/discussions/categories/polls)

Reply:
I am OK to leave it like this for now and archive #77.
But can we have test to make sure it only works on project?